File: C#/0922. Sort Array By Parity II.cs
public class Solution {
    public int[] SortArrayByParityII(int[] A) {
        int[] result=new int[A.Length];
        int even=1;
        int odd=0;
        foreach(int num in A){
            if(num%2==0){
                result[odd]=num;
                odd+=2;
            }
            else{
                result[even]=num;
                even+=2;
            }
        }
        return result;
    }
}

File: 20190519python/0922. Sort Array By Parity II.py
class Solution:
    def sortArrayByParityII(self, A: List[int]) -> List[int]:
        l=0
        r=len(A)-1
        while l<len(A) and r>0:
            while l<len(A) and A[l]%2==0:
                l+=2
            while r>=0 and A[r]%2!=0:
                r-=2
            if l<len(A):
                A[l],A[r]=A[r],A[l]
                l+=2
                r-=2
        return A
